src/NHibernate.Test/SystemTransactions/ResourceManagerFixture.cs

```diff
@@ -0,0 +1,889 @@
+					"Fake session opened, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				scope.Complete();
+				_log.Info("Scope completed");
+			}
+			_log.Info("Scope disposed");
+		}
+		
+		[Test]
+		public void DistributedNpgsqlCommit([Values(false, true)] bool enlistInPrepare)
+		{
+			using (var scope = new TransactionScope())
+			{
+				_log.InfoFormat(
+					"Scope opened, id {0}, distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate a Npgsql connection: as of Npgsql 3.2.4, volatile resource with single phase support
+				EnlistResource.EnlistVolatile(false, true);
+				_log.InfoFormat(
+					"Fake connection opened, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate another resource, not even supporting single phase (required for going distributed with "Npgsql")
+				EnlistResource.EnlistDurable();
+				_log.InfoFormat(
+					"Fake other resource, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate NHibernate : volatile no single phase support
+				if (enlistInPrepare)
+					EnlistResource.EnlistWithPrepareEnlistmentVolatile();
+				else
+					EnlistResource.EnlistVolatile();
+				_log.InfoFormat(
+					"Fake session opened, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				scope.Complete();
+				_log.Info("Scope completed");
+			}
+			_log.Info("Scope disposed");
+		}
+
+		#endregion
+
+		#region Failure
+
+		[Test]
+		[Explicit("Failing.")]
+		public void DistributedTransactionStatusMustBeInactiveAfterRollbackedScope()
+		{
+			SysTran transaction = null;
+			try
+			{
+				using (CreateDistributedTransactionScope())
+				{
+					_log.InfoFormat(
+						"Scope opened, id {0}, distributed id {1}",
+						SysTran.Current.TransactionInformation.LocalIdentifier,
+						SysTran.Current.TransactionInformation.DistributedIdentifier);
+					// The trouble occurs only with cloned transaction. The original one is disposed before and so
+					// considered inactive by FailsafeGetTransactionStatus test.
+					transaction = SysTran.Current.Clone();
+					_log.Info("Scope not completed");
+				}
+				_log.Info("Scope disposed");
+				Assert.That(FailsafeGetTransactionStatus(transaction), Is.Not.EqualTo(TransactionStatus.Active));
+			}
+			finally
+			{
+				transaction?.Dispose();
+			}
+		}
+
+		[Test]
+		[Explicit("Failing")]
+		public void DistributedTransactionFromCompletionEventShouldBeTheOneToWhichTheEventIsAttached()
```

Comment:
Here is that additional test demonstrating we do not receive the clone in transaction completion event. This one fails too when not distributed (see `TransactionFromCompletionEventShouldBeTheOneToWhichTheEventIsAttached`).